test/origin_simulator/origin_simulator_test.exs

```diff
@@ -11,242 +14,152 @@ defmodule OriginSimulatorTest do
+      |> OriginSimulator.call([])
+      |> assert_status_body(200, payload)
+      |> assert_resp_header({"content-type", ["application/json; charset=utf-8"]})
     end
 
     test "will return the headers in the payload when provided" do
-      payload = %{
-        "origin" => "https://www.bbc.co.uk/news",
-        "stages" => [%{ "at" => 0, "status" => 200, "latency" => "100ms..200ms"}],
-        "random_content" => nil,
-        "body"   => nil,
-        "headers" => %{"X-Foo" => "bar"}
-      }
+      payload = origin_recipe_headers() |> Poison.encode!()
+      conn(:post, "/add_recipe", payload) |> OriginSimulator.call([])
 
-      conn(:post, "/add_recipe", Poison.encode!(payload)) |> OriginSimulator.call([])
+      conn(:get, "/current_recipe")
+      |> OriginSimulator.call([])
+      |> assert_status_body(200, payload)
+      |> assert_resp_header({"content-type", ["application/json; charset=utf-8"]})
+    end
 
-      conn = conn(:get, "/current_recipe")
-      conn = OriginSimulator.call(conn, [])
+    # multiple recipes posting is currently not supported
+    test "/add_recipe returns error message if multiple recipes is posted" do
+      payload = multi_origin_payload() |> Poison.encode!()
 
-      assert conn.state == :sent
-      assert conn.status == 200
-      assert get_resp_header(conn, "content-type") == ["application/json; charset=utf-8"]
-      assert Poison.decode!(conn.resp_body) == payload
+      conn(:post, "/add_recipe", payload)
+      |> OriginSimulator.call([])
+      |> assert_status_body(406, "Not Acceptable")
+      |> assert_resp_header({"content-type", ["text/html; charset=utf-8"]})
     end
   end
 
-  describe "GET page with origin" do
+  describe "GET page" do
     test "will return the origin page" do
-      payload = %{
-        "origin" => "https://www.bbc.co.uk/news",
-        "stages" => [%{ "at" => 0, "status" => 200, "latency" => 0}]
-      }
-
-      conn(:post, "/add_recipe", Poison.encode!(payload)) |> OriginSimulator.call([])
-
-      Process.sleep 20
-
-      conn = conn(:get, "/")
-      conn = OriginSimulator.call(conn, [])
-
-      assert conn.state == :sent
-      assert conn.status == 200
-      assert get_resp_header(conn, "content-type") == ["text/html; charset=utf-8"]
-      assert conn.resp_body == "some content from origin"
+      payload = origin_recipe() |> Poison.encode!()
+      conn(:post, "/add_recipe", payload) |> OriginSimulator.call([])
+      Process.sleep(20)
+
+      conn(:get, "/")
+      |> OriginSimulator.call([])
+      |> assert_status_body(200, body_mock())
+      |> assert_resp_header({"content-type", ["text/html; charset=utf-8"]})
     end
-  end
 
-  describe "GET page with content" do
-    test "will return the passed content" do
-      payload = %{
-        "body" =>   "{\"hello\":\"world\"}",
-        "stages" => [%{ "at" => 0, "status" => 200, "latency" => 0}]
-      }
+    test "will return the origin page with random latency within range" do
```

Comment:
I think we're missing some detail in some of these test descriptions (not necessarily introduced in this PR of course). In this one for example, the description doesn't really us why there is random latency.